x-pack/plugin/ql/src/test/java/org/elasticsearch/xpack/ql/expression/predicate/RangeTests.java

```diff
@@ -0,0 +1,116 @@
+            { randomAlphaOfLength(10), randomFrom(K, T), d("2022-01-01"), DT, randomBoolean(), d("2021-01-01"), DT, randomBoolean(), true },
+            { randomAlphaOfLength(10), randomFrom(K, T), "now-10y", K, randomBoolean(), d("2022-01-01"), DT, randomBoolean(), false },
+            { randomAlphaOfLength(10), randomFrom(K, T), d("2021-01-01"), DT, true, "2021-01-01", K, true, false },
+            { randomAlphaOfLength(10), randomFrom(K, T), d("2021-01-01"), DT, false, "2021-01-01", K, true, true },
+            { randomAlphaOfLength(10), randomFrom(K, T), "2021-01-01", K, true, d("2021-01-01"), DT, false, true },
+            { randomAlphaOfLength(10), randomFrom(K, T), d("2021-01-01"), DT, false, d("2021-01-01"), DT, false, true },
+
+            // strings
+            {
+                randomAlphaOfLength(10),
+                randomFrom(K, T),
+                "a",
+                randomFrom(K, T),
+                randomBoolean(),
+                "b",
+                randomFrom(K, T),
+                randomBoolean(),
+                false },
+            {
+                randomAlphaOfLength(10),
+                randomFrom(K, T),
+                "b",
+                randomFrom(K, T),
+                randomBoolean(),
+                "a",
+                randomFrom(K, T),
+                randomBoolean(),
+                true },
+            { randomAlphaOfLength(10), randomFrom(K, T), "a", randomFrom(K, T), false, "a", randomFrom(K, T), false, true },
+
+            // numbers
+            { 10, randNType(), 1, randNType(), randomBoolean(), 10, randNType(), randomBoolean(), false },
+            { 10, randNType(), 10, randNType(), randomBoolean(), 1, randNType(), randomBoolean(), true },
+            { 10, randNType(), 1, randNType(), false, 1, randNType(), randomBoolean(), true },
+            { 10, randNType(), 1, randNType(), randomBoolean(), 1, randNType(), false, true },
+            { 10, randNType(), 1.0, randNType(), randomBoolean(), 10, randNType(), randomBoolean(), false },
+            { 10, randNType(), 1, randNType(), randomBoolean(), 10.D, randNType(), randomBoolean(), false },
+            { 10, randNType(), 10.0, randNType(), randomBoolean(), 1, randNType(), randomBoolean(), true },
+
+        };
+
+        for (int i = 0; i < tests.length; i++) {
+            Object[] test = tests[i];
+            Range range = new Range(
+                Source.EMPTY,
+                l(test[0], (DataType) test[1]),
+                l(test[2], (DataType) test[3]),
+                (Boolean) test[4],
+                l(test[5], (DataType) test[6]),
+                (Boolean) test[7],
+                ZoneId.systemDefault()
+            );
+            assertEquals("failed on test n " + i + ": " + Arrays.toString(test), test[8], range.areBoundariesInvalid());
+        }
+    }
+
+    private static Literal l(Object value, DataType type) {
```

Comment:
There's already a function having the same name and doing the same thing in `FunctionTestUtils.l`.